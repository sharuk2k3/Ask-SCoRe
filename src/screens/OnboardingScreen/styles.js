import {StyleSheet} from "react-native";
import {COLOR_LIGHT_PRIMARY} from "../../config/styles";

export default StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
  },

  buttonView: {
    marginHorizontal: 10,
    backgroundColor: COLOR_LIGHT_PRIMARY,
    width: 70,
    height: 30,
    borderRadius: 10,
    justifyContent: "center",
  },

  textView: {
    fontSize: 16,
    textAlign: "center",
  },

  titleView: {
    fontSize: 45,
    fontWeight: "200",
    marginBottom: 20,
  },

  subtitleView: {
    marginBottom: 50,
    letterSpacing: 2,
  },

  ImageView1: {
    width: 350,
    height: 200,
  },

  ImageView2: {
    width: 300,
    height: 276,
  },

  ImageView3: {
    width: 300,
    height: 180,
  },

  ImageView4: {
    width: 300,
    height: 250,
  },
});
