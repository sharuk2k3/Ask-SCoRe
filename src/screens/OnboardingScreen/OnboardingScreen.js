import React from "react";
import {View, Text, Image, TouchableOpacity} from "react-native";
import Onboarding from "react-native-onboarding-swiper";
import styles from "./styles";

const Dots = ({selected}) => {
  let backgroundColor = selected ? "#51AD28" : "#adadad";

  return (
    <View
      style={{
        width: 6,
        height: 6,
        marginHorizontal: 3,
        backgroundColor,
      }}
    />
  );
};

const Skip = ({...props}) => (
  <TouchableOpacity style={styles.buttonView} {...props}>
    <Text style={styles.textView}>Skip</Text>
  </TouchableOpacity>
);

const Next = ({...props}) => (
  <TouchableOpacity style={styles.buttonView} {...props}>
    <Text style={styles.textView}>Next</Text>
  </TouchableOpacity>
);

const Done = ({...props}) => (
  <TouchableOpacity style={styles.buttonView} {...props}>
    <Text style={styles.textView}>Done</Text>
  </TouchableOpacity>
);

const OnboardingScreen = ({navigation}) => {
  return (
    <Onboarding
      SkipButtonComponent={Skip}
      NextButtonComponent={Next}
      DoneButtonComponent={Done}
      DotComponent={Dots}
      onSkip={() => navigation.replace("Start")}
      onDone={() => navigation.navigate("Start")}
      pages={[
        {
          backgroundColor: "#fff",
          image: (
            <Image
              style={styles.ImageView1}
              source={require("../../../assets/OB1.png")}
            />
          ),
          title: <Text style={styles.titleView}>Community</Text>,
          subtitle: (
            <Text style={{marginBottom: 50, letterSpacing: 2}}>
              Support and Help Each Other
            </Text>
          ),
        },
        {
          backgroundColor: "#fff",
          image: (
            <Image
              style={styles.ImageView2}
              source={require("../../../assets/OB2.png")}
            />
          ),
          title: <Text style={styles.titleView}>Collaboration</Text>,
          subtitle: (
            <Text style={styles.subtitleView}>
              Collaborate and sync on Projects
            </Text>
          ),
        },
        {
          backgroundColor: "#fff",
          image: (
            <Image
              style={styles.ImageView3}
              source={require("../../../assets/OB3.png")}
            />
          ),
          title: <Text style={styles.titleView}>Communicate</Text>,
          subtitle: (
            <Text style={styles.subtitleView}>
              Communication is what makes team strong
            </Text>
          ),
        },
        {
          backgroundColor: "#fff",
          image: (
            <Image
              style={styles.ImageView4}
              source={require("../../../assets/OB4.png")}
            />
          ),
          title: <Text style={styles.titleView}>Queries</Text>,
          subtitle: (
            <Text style={styles.subtitleView}>Get your queries Resolved</Text>
          ),
        },
      ]}
    />
  );
};

export default OnboardingScreen;
