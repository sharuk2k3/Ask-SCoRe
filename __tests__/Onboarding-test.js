import React from "react";
import OnboardingScreen from "../src/screens/OnboardingScreen/OnboardingScreen";
import renderer from "react-test-renderer";

jest.useFakeTimers();

test("renders Question Screen correctly", () => {
  const tree = renderer.create(<OnboardingScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
