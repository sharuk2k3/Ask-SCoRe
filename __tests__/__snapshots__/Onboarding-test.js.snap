// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Question Screen correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "backgroundColor": "#fff",
          "image": <Image
            source="test-file-stub"
            style={
              Object {
                "height": 200,
                "width": 350,
              }
            }
          />,
          "subtitle": <Text
            style={
              Object {
                "letterSpacing": 2,
                "marginBottom": 50,
              }
            }
          >
            Support and Help Each Other
          </Text>,
          "title": <Text
            style={
              Object {
                "fontSize": 45,
                "fontWeight": "200",
                "marginBottom": 20,
              }
            }
          >
            Community
          </Text>,
        },
        Object {
          "backgroundColor": "#fff",
          "image": <Image
            source="test-file-stub"
            style={
              Object {
                "height": 276,
                "width": 300,
              }
            }
          />,
          "subtitle": <Text
            style={
              Object {
                "letterSpacing": 2,
                "marginBottom": 50,
              }
            }
          >
            Collaborate and sync on Projects
          </Text>,
          "title": <Text
            style={
              Object {
                "fontSize": 45,
                "fontWeight": "200",
                "marginBottom": 20,
              }
            }
          >
            Collaboration
          </Text>,
        },
        Object {
          "backgroundColor": "#fff",
          "image": <Image
            source="test-file-stub"
            style={
              Object {
                "height": 180,
                "width": 300,
              }
            }
          />,
          "subtitle": <Text
            style={
              Object {
                "letterSpacing": 2,
                "marginBottom": 50,
              }
            }
          >
            Communication is what makes team strong
          </Text>,
          "title": <Text
            style={
              Object {
                "fontSize": 45,
                "fontWeight": "200",
                "marginBottom": 20,
              }
            }
          >
            Communicate
          </Text>,
        },
        Object {
          "backgroundColor": "#fff",
          "image": <Image
            source="test-file-stub"
            style={
              Object {
                "height": 250,
                "width": 300,
              }
            }
          />,
          "subtitle": <Text
            style={
              Object {
                "letterSpacing": 2,
                "marginBottom": 50,
              }
            }
          >
            Get your queries Resolved
          </Text>,
          "title": <Text
            style={
              Object {
                "fontSize": 45,
                "fontWeight": "200",
                "marginBottom": 20,
              }
            }
          >
            Queries
          </Text>,
        },
      ]
    }
    disableVirtualization={false}
    extraData={null}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    initialNumToRender={1}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    onViewableItemsChanged={[Function]}
    pagingEnabled={true}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsHorizontalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfig={
      Object {
        "itemVisiblePercentThreshold": 100,
      }
    }
    viewabilityConfigCallbackPairs={
      Array [
        Object {
          "onViewableItemsChanged": [Function],
          "viewabilityConfig": Object {
            "itemVisiblePercentThreshold": 100,
          },
        },
      ]
    }
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "center",
                "paddingTop": 0,
              },
              null,
              Object {
                "height": 1334,
                "width": 750,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 0,
                  "paddingBottom": 60,
                  "width": "100%",
                },
                null,
              ]
            }
          >
            <Image
              source="test-file-stub"
              style={
                Object {
                  "height": 200,
                  "width": 350,
                }
              }
            />
          </View>
          <Text
            style={
              Object {
                "fontSize": 45,
                "fontWeight": "200",
                "marginBottom": 20,
              }
            }
          >
            Community
          </Text>
          <Text
            style={
              Object {
                "letterSpacing": 2,
                "marginBottom": 50,
              }
            }
          >
            Support and Help Each Other
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "width": 0,
          }
        }
      />
    </View>
  </RCTScrollView>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.1)",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flexDirection": "row",
          "height": 60,
          "justifyContent": "space-between",
          "paddingHorizontal": 0,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flexShrink": 1,
            "width": 200,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#9cd683",
              "borderRadius": 10,
              "height": 30,
              "justifyContent": "center",
              "marginHorizontal": 10,
              "opacity": 1,
              "width": 70,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "textAlign": "center",
              }
            }
          >
            Skip
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 0,
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#51AD28",
              "height": 6,
              "marginHorizontal": 3,
              "width": 6,
            }
          }
        />
        <View
          style={
            Object {
              "backgroundColor": "#adadad",
              "height": 6,
              "marginHorizontal": 3,
              "width": 6,
            }
          }
        />
        <View
          style={
            Object {
              "backgroundColor": "#adadad",
              "height": 6,
              "marginHorizontal": 3,
              "width": 6,
            }
          }
        />
        <View
          style={
            Object {
              "backgroundColor": "#adadad",
              "height": 6,
              "marginHorizontal": 3,
              "width": 6,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "flexShrink": 1,
            "width": 200,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#9cd683",
              "borderRadius": 10,
              "height": 30,
              "justifyContent": "center",
              "marginHorizontal": 10,
              "opacity": 1,
              "width": 70,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "textAlign": "center",
              }
            }
          >
            Next
          </Text>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;
